drivers/sensor/kx022/kx022_trigger.c

```diff
@@ -0,0 +1,445 @@
+/* Kionix KX022 3-axis accelerometer driver
+ *
+ * Copyright (c) 2021 G-Technologies Sdn. Bhd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <stdio.h>
+#include <device.h>
+#include <drivers/i2c.h>
+#include <sys/__assert.h>
+#include <sys/util.h>
+#include <kernel.h>
+#include <drivers/sensor.h>
+#include <logging/log.h>
+#include "kx022.h"
+
+#if KX022_DEFAULT_INT_PIN == KX022_INT2
+#error INT2 not supported
+#endif
+
+LOG_MODULE_DECLARE(KX022, CONFIG_SENSOR_LOG_LEVEL);
+
+static void kx022_gpio_callback(const struct device *dev, struct gpio_callback *cb, uint32_t pins)
+{
+	struct kx022_data *data = CONTAINER_OF(cb, struct kx022_data, gpio_cb);
+	
+#if defined(CONFIG_KX022_TRIGGER_OWN_THREAD)
+	k_sem_give(&data->trig_sem);
+#elif defined(CONFIG_KX022_TRIGGER_GLOBAL_THREAD)
+	k_work_submit(&data->work);
+#endif
+}
+
+static void kx022_handle_motion_int(const struct device *dev)
+{
+	struct kx022_data *data = dev->data;
+
+	if (data->motion_handler != NULL) {
+		data->motion_handler(dev, &data->motion_trigger);
+	}
+}
+
+static void kx022_handle_drdy_int(const struct device *dev)
+{
+	struct kx022_data *data = dev->data;
+
+	if (data->drdy_handler != NULL) {
+		data->drdy_handler(dev, &data->drdy_trigger);
+	}
+}
+static void kx022_handle_slope_int(const struct device *dev)
+{
+	struct kx022_data *data = dev->data;
+
+	if (data->slope_handler != NULL) {
+		data->slope_handler(dev, &data->slope_trigger);
+	}
+}
+
+static void kx022_handle_int(const struct device *dev)
+{
+	struct kx022_data *data = dev->data;
+	uint8_t status, clr;
+
+	if (data->hw_tf->read_reg(data, KX022_REG_INS2, &status) < 0) {
+		return;
+	}
+
+	if (status & KX022_MASK_INS2_WUFS) {
+		kx022_handle_motion_int(dev);
+	}
+
+	if (status & KX022_MASK_INS2_DRDY) {
+		kx022_handle_drdy_int(dev);
+		//k_msleep(50);
```

Comment:
?

Reply:
azam old code,
not use now,
need remove?